libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/ModelState.hs

```diff
@@ -0,0 +1,443 @@
+pParamsZeroByProof (Alonzo _) = def
+pParamsZeroByProof (Mary _) = def
+pParamsZeroByProof (Allegra _) = def
+pParamsZeroByProof (Shelley _) = def
+
+pPUPStateZero :: forall era. Reflect era => State (Core.EraRule "PPUP" era)
+pPUPStateZero = pPUPStateZeroByProof @era (reify :: Proof era)
+
+uTxOStateZero :: forall era. Reflect era => UTxOState era
+uTxOStateZero = smartUTxOState utxoZero mempty mempty (pPUPStateZero @era)
+
+pParamsZero :: Reflect era => Core.PParams era
+pParamsZero = lift pParamsZeroByProof
+
+ledgerStateZero :: Reflect era => LedgerState era
+ledgerStateZero = LedgerState uTxOStateZero dPStateZero
+
+epochStateZero :: Reflect era => EpochState era
+epochStateZero = EpochState accountStateZero snapShotsZero ledgerStateZero pParamsZero pParamsZero nonMyopicZero
+
+newEpochStateZero :: forall era. Reflect era => NewEpochState era
+newEpochStateZero =
+  NewEpochState
+    (EpochNo 0)
+    blocksMadeZero
+    blocksMadeZero
+    epochStateZero
+    SNothing
+    poolDistrZero
+    (stashedAVVMAddressesZero (reify :: Proof era))
+
+stashedAVVMAddressesZero :: Proof era -> StashedAVVMAddresses era
+stashedAVVMAddressesZero (Shelley _) = utxoZero
+stashedAVVMAddressesZero (Babbage _) = ()
+stashedAVVMAddressesZero (Alonzo _) = ()
+stashedAVVMAddressesZero (Mary _) = ()
+stashedAVVMAddressesZero (Allegra _) = ()
+
+mNewEpochStateZero :: Reflect era => ModelNewEpochState era
+mNewEpochStateZero =
+  ModelNewEpochState
+    { mPoolParams = Map.empty,
+      mRewards = Map.empty,
+      mDelegations = Map.empty,
+      mUTxO = Map.empty,
+      mMutFee = Map.empty,
+      mAccountState = accountStateZero,
+      mPoolDistr = Map.empty,
+      mPParams = pParamsZero,
+      mdeposited = Coin 0,
+      mfees = Coin 0,
+      mCount = 0,
+      mIndex = Map.empty,
+      -- below here NO EFFECT until we model EpochBoundary
+      mFPoolParams = Map.empty,
+      mRetiring = Map.empty,
+      mSnapshots = snapShotsZero,
+      mEL = EpochNo 0,
+      mBprev = Map.empty,
+      mBcur = Map.empty,
+      mRu = SNothing
+    }
+
+testNES :: NewEpochState (BabbageEra Mock)
+testNES = newEpochStateZero
+
+testMNES :: ModelNewEpochState (BabbageEra Mock)
+testMNES = mNewEpochStateZero
+
+-- ======================================================================
+
+class Extract t era where
+  extract :: ModelNewEpochState era -> t
+
+instance Crypto era ~ c => Extract (DState c) era where
+  extract x =
+    DState
+      (UMap.unify (mRewards x) (mDelegations x) Map.empty)
+      Map.empty
+      genDelegsZero
+      instantaneousRewardsZero
+
+instance Crypto era ~ c => Extract (PState c) era where
+  extract x = PState (mPoolParams x) (mFPoolParams x) (mRetiring x)
+
+instance Crypto era ~ c => Extract (DPState c) era where
+  extract x = DPState (extract x) (extract x)
+
+instance Reflect era => Extract (UTxOState era) era where
+  extract x = smartUTxOState (UTxO (SplitMap.fromMap (mUTxO x))) (mdeposited x) (mfees x) (pPUPStateZero @era)
```

Comment:
This is quite a long line. Rule of thumb for me is if I can't see the end of a line in a PR it is already too long
```suggestion
  extract x =
    smartUTxOState 
    (UTxO (SplitMap.fromMap (mUTxO x))) (mdeposited x) (mfees x) (pPUPStateZero @era)
```

Reply:
done